Changelog for the 0.14.0 release

jrfnl opened the pull request:
Complete, including the open PRs #1213 and #1214.

If there are no objections and no pertinent bugs found between now and then, I propose to release 0.14.0 ~~this Thursday~~ ~~this Monday (see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1215#issuecomment-339552840)~~ this Wednesday (delayed for reasons known to the team).

GaryJones commented on CHANGELOG.md:
```diff
@@ -8,6 +8,65 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
+- The `is_class_constant()`, `is_class_property` and `valid_direct_scope()` utility methods to the `WordPress\Sniff` class.
+
+### Changed
+- When passing an array property via a custom ruleset to PHP Codesniffer, spaces around the key/value are taken as intentional and parsed as part of the array key/value. In practice, this leads to confusion and WPCS does not expect any values which could be preceded/followed by a space, so for the WordPress Coding Standard native array properties, like `customAutoEscapedFunction`, `text_domain`,  ...
```
"PHP Codesniffer" => "PHP_CodeSniffer"

jrfnl replied:
Just wondering: AFAICS the package name is "PHP_Codesniffer", but the product name is "PHP Codesniffer", so why should the package name be used here instead of the product name ? or am I missing something ?

GaryJones replied:
The product name is also called `PHP_CodeSniffer` (underscore, capital S) - see the PHPCS [Readme](https://github.com/squizlabs/PHP_CodeSniffer), the PHPCS [wiki](https://github.com/squizlabs/PHP_CodeSniffer/wiki), the SquizLabs [blog posts](https://www.squizlabs.com/php-codesniffer/), `phpcs --version` etc. 

The package is `squizlabs/php_codesniffer`, but everywhere else official it is referred to as exactly `PHP_CodeSniffer`, and not `PHP Codesniffer`.

jrfnl replied:
Okidoki. Changed all pertinent references to PHPCS which were misspelled (in Readme/Changelog/Contributing).

Interestingly enough, in some code which is literally taken from PHPCS itself (test suite files for 2.x test support), the name is set as "PHP CodeSniffer"... 
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/615323e260492c6192bda208d390dab7158aa937/Test/AllTests.php#L53
* https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/615323e260492c6192bda208d390dab7158aa937/Test/Standards/AllSniffs.php#L58

As those are copied-in files, I haven't changed those.

GaryJones commented on CHANGELOG.md:
```diff
@@ -8,6 +8,65 @@ This projects adheres to [Semantic Versioning](http://semver.org/) and [Keep a C
+- Improved the Atom setup instructions in the Readme.
+- Updated the unit testing information in Contributing.
+- Updated the [custom ruleset example](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/phpcs.xml.dist.sample) for the changes contained in this release and to make it more explicit what is recommended versus example code.
+- The minimum recommended version for the suggested `DealerDirect/phpcodesniffer-composer-installer` Composer plugin has gone up to `0.4.3`.
```
...to address a [bug](link).

jrfnl replied:
OK if I just add " _This patch version fixes support for PHP 5.3._"

GaryJones replied:
Yep - just needs a clarification as to why we bumped the minimum recommended version.

jrfnl replied:
:+1: Done.

GaryJones commented on README.md:
```diff
@@ -223,7 +223,7 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
 ### Visual Studio
 
-Please see "[Setting up PHP CodeSniffer in Visual Studio Code](https://tommcfarlin.com/php-codesniffer-in-visual-studio-code/)", a tutorial by Tom McFarlin.
+Please see "[Setting up PHP_CodeSniffer in Visual Studio Code](https://tommcfarlin.com/php-codesniffer-in-visual-studio-code/)", a tutorial by Tom McFarlin.
```
As the title of @tommcfarlin's post, this one shouldn't change (unless Tom is happy to change references in his post).

jrfnl replied:
Reverted this one.

jrfnl commented:
As #1209 got merged, I've added a new entry to the changelog for it (separate commit, please squash the commits on merge).

It's sister-PR #1166 should probably also still go into this release and I'd be happy for #1216 to go in as well.

To that end, I've updated the projected release date to October 30 (this Monday).

If both this PR as well as #1165 got the necessary approvals by Monday and there are no objections or pertinent bugs between now and then, I will merge both this Monday and release.

jrfnl commented:
Updated in a new commit for easier review: updated release date, added entry about PR #1217 and made a minor adjustment to the entry about the `PreparedSQLPlaceholders` sniff.

jrfnl commented:
Right, let's go for it!
